shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -3189,7 +3201,10 @@ module MakeImpl<InputSig Lang> {
       Typ getTyp(DataFlowType t) { result = t }
 
       bindingset[c, t, tail]
-      Ap apCons(Content c, Typ t, Ap tail) { result.isCons(c, t, tail) }
+      Ap apCons(Content c, Typ t, Ap tail) {
+        result.isCons(c, t, tail) and
+        Config::accessPathLimit() > tail.len()
```

Comment:
This should not be necessary - just ensure that all `AccessPathApprox` satisfy the limit.

Reply:
Most do, you just need the `> 0` check in the `ConsNil` case, I think.

Reply:
Right.